src/main/java/org/broadinstitute/consent/http/service/LibraryCardService.java

```diff
@@ -113,7 +113,11 @@ public List<LibraryCard> addDaaToUserLibraryCardByInstitution(User user, Integer
         .toList();
     // typically there should be one library card per user per institution
     for (LibraryCard libraryCard : matchingLibraryCards) {
-      addDaaToLibraryCard(libraryCard.getId(), daaId);
+      try {
+        addDaaToLibraryCard(libraryCard.getId(), daaId);
```

Comment:
Another thing we can do to tweak the sql is to do nothing if the join record exists. This method in LibraryCardDAO:
```
  @SqlUpdate("""
      INSERT INTO lc_daa (lc_id, daa_id)
      VALUES (:lcId, :daaId)
      """)
  void createLibraryCardDaaRelation(@Bind("lcId") Integer lcId, @Bind("daaId") Integer daaId);
```
Could possibly be:
```
  @SqlUpdate("""
      INSERT INTO lc_daa (lc_id, daa_id)
      VALUES (:lcId, :daaId)
      ON CONFLICT DO NOTHING
      """)
  void createLibraryCardDaaRelation(@Bind("lcId") Integer lcId, @Bind("daaId") Integer daaId);
```

Reply:
That's probably a better solution. I'll do that instead.

Reply:
Fixed in https://github.com/DataBiosphere/consent/pull/2308/commits/6a25bf822186af72d85568e30d9fa56d5d90a214